File: scsc-monolith/src/main/java/demo/scsc/config/JpaEntityManagerProvider.java
package demo.scsc.config;

import jakarta.persistence.EntityManager;
import org.axonframework.common.jpa.EntityManagerProvider;

public class JpaEntityManagerProvider implements EntityManagerProvider {
    private final JpaPersistenceUnit jpaPersistence;

    JpaEntityManagerProvider(JpaPersistenceUnit jpaPersistence) {
        this.jpaPersistence = jpaPersistence;
    }

    @Override
    public EntityManager getEntityManager() {
        return jpaPersistence.getThreadLocalEntityManager();
    }
}

File: scsc-monolith/src/main/kotlin/demo/scsc/config/JpaEntityManagerProvider.kt
package demo.scsc.config

import jakarta.persistence.EntityManager
import org.axonframework.common.jpa.EntityManagerProvider

class JpaEntityManagerProvider internal constructor(private val jpaPersistence: JpaPersistenceUnit) :
    EntityManagerProvider {
    override fun getEntityManager(): EntityManager? = jpaPersistence.threadLocalEntityManager
}
